File: C_Programmes/Matrix_Operations.c
//C program to perform the matrix operations addition, subtraction and multiplication according to the users preference

#include<stdio.h>
void Add(int r1, int c1, int r2, int c2, int [r1][c1], int [r2][c2]);
void Subtract(int r1, int c1, int r2, int c2, int [r1][c1], int [r2][c2]);
void Multiply(int r1, int c1, int r2, int c2, int [r1][c1], int [r2][c2]);

int main()
{
    int r1, c1, r2, c2;
    int choice = 0;
    int loop = 1;
    while(loop)
    {
        printf("Enter the number of rows of the first matrix\n");
        scanf("%d",&r1);
        printf("Enter the number of columns of the first matrix\n");
        scanf("%d",&c1);
        int Array1[r1][c1];
        printf("Enter the array elements\n");
        for (int i = 0; i<r1; i++)
        {
            for(int j = 0; j<c1; j++)
            {
                scanf("%d",&Array1[i][j]);
            }
        }
        printf("Enter the number of rows of the second matrix\n");
        scanf("%d",&r2);
        printf("Enter the number of columns of the second matrix\n");
        scanf("%d",&c2);
        int Array2[r2][c2];
        printf("Enter the array elements\n");
        for (int i = 0; i<r2; i++)
        {
            for(int j = 0; j<c2; j++)
            {
                scanf("%d",&Array2[i][j]);
            }
        }
        printf("Enter 1 for Matrtix addition\n2 for Matrix Subtraction\n3 for Matrix Multiplication\n4 to Quit\n");
        scanf("%d",&choice);
        switch(choice)
        {
            case 1:
                Add(r1,c1,r2,c2,Array1, Array2);
                break;
            case 2:
                Subtract(r1,c1,r2,c2,Array1, Array2);
                break;
            case 3:
                Multiply(r1,c1,r2,c2,Array1,Array2);
                break;
            case 4:
                loop = 0;
                break;
            default:
                printf("Invalid entry\nRetry\n");
                break;
        }
    }
    return 0;
}

void Add(int r1, int c1, int r2, int c2, int Array1[r1][c1], int Array2[r2][c2] )
{
    if(r1==r2 && r2==c2)
    {
        int Result[r1][c1];
        printf("The addition is:\n");
        for (int i = 0; i<r1; i++)
            {
                for(int j = 0; j<c1; j++)
                {
                    Result[i][j] = 0;
                    Result[i][j] = Array1[i][j] + Array2[i][j];
                    printf("%d ",Result[i][j]);
                }
                printf("\n");
            }
        return;
    }
    else
    {
        printf("Matrix addition is not possible\n");
        return;
    }

}

void Subtract(int r1, int c1, int r2, int c2, int Array1[r1][c1], int Array2[r2][c2] )
{
    if(r1==r2 && r2==c2)
    {
        int Result[r1][c1];
        printf("The subtraction is:\n");
        for (int i = 0; i<r1; i++)
            {
                for(int j = 0; j<c1; j++)
                {
                    Result[i][j] = 0;
                    Result[i][j] = Array1[i][j] - Array2[i][j];
                    printf("%d ",Result[i][j]);
                }
                printf("\n");
            }
        return;
    }
    else
    {
        printf("Matrix subtraction is not possible\n");
        return;
    }

}

void Multiply(int r1, int c1, int r2, int c2, int Array1[r1][c1], int Array2[r2][c2] )
{
    if(r2==c1)
    {
        int Result[r1][c2];
        printf("The multiplication is:\n");
        for (int i = 0; i<r1; i++)
            {
                for(int j = 0; j<c2; j++)
                {
                    Result[i][j] = 0;
                    for (int k=0;k<r2;k++)
                    {
                        Result[i][j] += Array1[i][k] * Array2[k][j];
                    }
                    printf("%d ",Result[i][j]);
                }
                printf("\n");
            }
    }
    else
    {
        printf("Matrix multiplication is not possible\n");
    }

}

File: Java_Programmes/Matrix_Operations.java
import java.util.Scanner;
public class Matrix_Operations {
    public static void main(String [] args)
    {
        Scanner in = new Scanner(System.in);
        int r1, c1, r2, c2;
        int choice = 0;
        int loop = 1;
        while(loop==1)
        {
            System.out.println("Enter the number of rows of the first matrix");
            r1 = in.nextInt();
            System.out.println("Enter the number of columns of the first matrix");
            c1 = in.nextInt();
            int [][] Array1 =new int [r1][c1];
            System.out.println("Enter the array elements");
            for (int i = 0; i<r1; i++)
            {
                for(int j = 0; j<c1; j++)
                {
                    Array1[i][j] = in.nextInt();
                }
            }
            System.out.println("Enter the number of rows of the second matrix");
            r2 = in.nextInt();
            System.out.println("Enter the number of columns of the second matrix");
            c2 = in.nextInt();
            int [][] Array2 =new int [r2][c2];
            System.out.println("Enter the array elements");
            for (int i = 0; i<r2; i++)
            {
                for(int j = 0; j<c2; j++)
                {
                    Array2[i][j] = in.nextInt();
                }
            }
            System.out.println("Enter 1 for Matrix addition\n2 for Matrix Subtraction\n3 for Matrix Multiplication\n4 to Quit");
            choice = in.nextInt();
            switch(choice)
            {
                case 1:
                    Add(r1,c1,r2,c2,Array1, Array2);
                    break;
                case 2:
                    Subtract(r1,c1,r2,c2,Array1, Array2);
                    break;
                case 3:
                    Multiply(r1,c1,r2,c2,Array1,Array2);
                    break;
                case 4:
                    loop = 0;
                    break;
                default:
                    System.out.println("Invalid entryRetry");
                    break;
            }
        }
        in.close();

    }
    public static void Add(int r1, int c1, int r2, int c2, int [][] Array1, int [][] Array2 )
    {
        if(r1==r2 && r2==c2)
        {
            int [][] Result =new int [r1][c1];
            System.out.println("The addition is:");
            for (int i = 0; i<r1; i++)
                {
                    for(int j = 0; j<c1; j++)
                    {
                        Result[i][j] = 0;
                        Result[i][j] = Array1[i][j] + Array2[i][j];
                        System.out.print(Result[i][j]+" ");
                    }
                    System.out.println("");
                }
            return;
        }
        else
        {
            System.out.println("Matrix addition is not possible");
            return;
        }

    }

    public static void Subtract(int r1, int c1, int r2, int c2, int [][] Array1, int [][] Array2 )
    {
        if(r1==r2 && r2==c2)
        {
            int [][] Result =new int [r1][c1];
            System.out.println("The subtraction is:");
            for (int i = 0; i<r1; i++)
                {
                    for(int j = 0; j<c1; j++)
                    {
                        Result[i][j] = 0;
                        Result[i][j] = Array1[i][j] - Array2[i][j];
                        System.out.print(Result[i][j]+" ");
                    }
                    System.out.println("");
                }
            return;
        }
        else
        {
            System.out.println("Matrix subtraction is not possible");
            return;
        }

    }

    public static void Multiply(int r1, int c1, int r2, int c2, int  [][] Array1, int [][] Array2 )
    {
        if(r2==c1)
        {
            int [][] Result =new int [r1][c2];
            System.out.println("The multiplication is:");
            for (int i = 0; i<r1; i++)
                {
                    for(int j = 0; j<c2; j++)
                    {
                        Result[i][j] = 0;
                        for (int k=0;k<r2;k++)
                        {
                            Result[i][j] += Array1[i][k] * Array2[k][j];
                        }
                        System.out.print(Result[i][j]+" ");
                    }
                    System.out.println("");
                }
        }
        else
        {
            System.out.println("Matrix multiplication is not possible");
        }

    }
}
